.github/workflows/main-pr.yaml

```diff
@@ -20,6 +20,11 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: setup golang
+        uses: actions/setup-go@v3
+        with:
+          go-version: '^1.20'
```

Comment:
Given that this introduces a separate, independent lint step I'd extract it out to a separate job.

And while we're here I suggest that we add go module caching like in other Github Workflows for Go projects that we use.

Reply:
Done